lib/logger_json/plug.ex

```diff
@@ -31,16 +33,28 @@ if Code.ensure_loaded?(Plug) do
       level = Keyword.get(opts, :log, :info)
       client_version_header = Keyword.get(opts, :version_header, "x-api-version")
       metadata_formatter = Keyword.get(opts, :metadata_formatter, MetadataFormatters.GoogleCloudLogger)
-      {level, metadata_formatter, client_version_header}
+
+      extra_attributes_fn =
+        case Keyword.get(opts, :extra_attributes_fn) do
+          fun when is_function(fun) ->
+            fun
+
+          nil ->
+            nil
+            # Raise for anything else or set as nil?
```

Comment:
We definitely should raise here, otherwise, you can rely on this method being called but due to some mistake, it never will, silently.

Reply:
Done.